tests/tape/test_tape.py

```diff
@@ -844,7 +844,7 @@ def test_expand_tape_multiple_wires_non_commuting(self, ret):
     def test_is_sampled_reserved_after_expansion(self, monkeypatch, mocker):
         """Test that the is_sampled property is correctly set when tape
         expansion happens."""
-        dev = qml.device('default.qubit', wires=1, analytic=True)
```

Comment:
Was this a bug? How come `is_sampled` was true when analytic was set to `True`? Shouldn't it have been `False`? 🤔

Reply:
Another case of how the old logic was confusing :) 

The test uses `sample` as outputs. Now we cannot do this any more on an analytic device, so the test case needs to change...

Reply:
Oh, I see. I feel like this PR will be clearing up a lot of confusion. Great! 🙂